src/mem/cache/prefetch/stride.cc

```diff
@@ -106,14 +106,15 @@ Stride::allocateNewContext(int context)
 {
     // Create new table
     auto insertion_result = pcTables.insert(std::make_pair(context,
-        PCTable(pcTableInfo.assoc, pcTableInfo.numEntries,
-        pcTableInfo.indexingPolicy, pcTableInfo.replacementPolicy,
-        StrideEntry(initConfidence))));
+	new PCTable("PCTable", pcTableInfo.numEntries, pcTableInfo.assoc,
```

Comment:
I am not sure why, but the destructor for PCTable was being called without this 'new'. Any ideas why? How was it working before?

Reply:
Because pcTables is a map of <int, PCTable>. It is not a pointer, so it shouldn´t be newed.